src/pmdas/bpf/bpf.conf

```diff
@@ -0,0 +1,3 @@
+# list the modules to start here
```

Comment:
We gonna stick with this config file or make the agent completely dynamic?  (this modules list can be built via a directory scan - easier install / maintenance)

If module configuration becomes necessary for some modules (in future) module-specific configs might be simpler.

Reply:
+1

I'd go with `mods-enabled` and `mods-available` directories, where users can move (or symlink) scripts from the `mods-available` directory to the `mods-enabled` directory. The `mods-enabled` directory and all files inside must be exclusively writeable by root, and all modules inside will be loaded on startup.

Regarding module configuration: Users could write sections like
```
[tracing.so]
process_name = firefox
```
in `bpf.conf`, to overwrite any default module configuration values.

I'm not sure if we need the flexibility of running two instances of the same module (with different configs) - for that we'd need another solution (the bcc PMDA can do that for example).

Reply:
Agree generally a configuration approach is needed.

I'm not sure I understand the benefit to the approach of enumerating directory to load, feels like we are splitting configuration across two places (module directory and `bpf.conf`. How do you see the benefit vs listing the modules to load directly in the config file as bcc does now .. maybe something learned in bcc development that I'm missing?

Reply:
> We gonna stick with this config file or make the agent completely dynamic? (this modules list can be built via a directory scan - easier install / maintenance)
> If module configuration becomes necessary for some modules (in future) module-specific configs might be simpler.

Yeah, my initial thinking was let's start with a basic configuration file parser. I'm not clear on the benefit of enumerating directories over listing in the file, as a user likely has to configure the file anyway?

One alternative would be the module is responsible for its own configuration somehow.

Reply:
I've introduced a configuration file and use of `iniparser`. Nothing _uses_ it yet, but the capability is there and a module can pick it up and read it.